lib/routes/account.js

```diff
@@ -1911,8 +1982,199 @@ module.exports = function (
             function () {
               reply({})
             },
+            function (err) {
+              reply(err)
+            }
+          )
+      }
+    },
+    {
+      method: 'GET',
+      path: '/recovery_emails',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        response: {
+          schema: isA.array().items(
+            isA.object({
+              verified: isA.boolean().required(),
+              isPrimary: isA.boolean().required(),
+              email: validators.email().required()
+            }))
+        }
+      },
+      handler: function (request, reply) {
+        const sessionToken = request.auth.credentials
+        const uidHex = sessionToken.uid.toString('hex')
+        var emails
+
+        log.begin('Account.RecoveryEmailEmails', request)
+
+        return getAccountEmails()
+          .then(createResponse)
+          .done(reply, reply)
+
+        function getAccountEmails() {
+          return db.accountEmails(uidHex)
+            .then(function (result) {
+              emails = result
+            })
+        }
+
+        function createResponse() {
```

Comment:
Again, no biggie but `createResponse` could take `emails` as an argument and we could eliminate setting the `var` in the parent scope.